### DISCUSSION TITLE
Kotlinx Serialization

### DISCUSSION THREAD

**Discussion by jhouse-pm** [mrclrchtr/gradle-kotlin-spring#105] [Q&A] [answered]
Thanks for the template, I'm new to SpringBoot and this is a great resource!

I'm having issues added Kotlin's serialization library to buildSrc and thought to ask for assistance:

I've tried adding:
``` kotlin
// buildSrc/.../kotlin-conventions.gradle.kts
plugins {
    kotlin("plugin.serialization")
}
```

but am seeing the following error:

```
FAILURE: Build failed with an exception.

* Where:
Precompiled script plugin '/Users/jhouse/repos/kt-landslide/buildSrc/src/main/kotlin/com/pavlovmedia/landside/gradle/kotlin-conventions.gradle.kts' line: 1

* What went wrong:
Plugin [id: 'org.jetbrains.kotlin.plugin.serialization'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Plugin Repositories (plugin dependency must include a version number for this source)
```

I'm hoping once I solve that then I can the follow dependency to either a) all sub-projects (in kotlin-conventions.gradle.kts??), or b) specific sub-projects:

``` kotlin
dependency {
    implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:1.6.0")
}
```

Thanks for any assistance in how this lib can be setup in your template.

**Comment by mrclrchtr** [comment, marked as answer]
You need to add the dependency to `buildSrc/build.gradle.kts`:
```
add("implementation", libs.findLibrary("kotlin-serialization").get())
```
AND in `gradle/libs.versions.toml` under `libraries`:
```
kotlin-serialization = { module = "org.jetbrains.kotlin:kotlin-serialization", version.ref = "kotlin" }
```

**Reply by jhouse-pm** [reply]
Thanks for the reply. That did the trick!
